### DISCUSSION TITLE
How to set a colour based on a environment variable?

### DISCUSSION THREAD

**Discussion by mbrasch** [starship/starship#5173] [General]
I have 3 shell environments, here AAAA, BBBB and CCCC. When I am in an environment, $henv is set to one of these 3 values. I would now like to have these environment strings displayed in the prompt. AAAA with a red background, BBBB with a yellow background and CCCC with a green background.

I've been trying around here for almost 2 hours now and don't understand how I can do this - help... :)

**Comment by mbrasch** [comment]
Ah, forgotten: I tried it with this code fragment:

```toml
[custom.henv]
command = """henv_value=$(echo $henv)
case $henv_value in
AAAA)
  echo -ne "bg:red"
  ;;
BBBB)
  echo -ne "bg:green"
  ;;
CCCC)
  echo -n "bg:yellow"
  ;;
*)
  echo -n ""
  ;;
esac
"""
style = "($output)"
when = true
format = "[$output]($output)"
```

I output $output here to see what the result of command was. Normally $henv_value should be output here.

**Reply by mickimnet** [reply]
`$output` doesn't work as _style_. What is the actual string you want to display?

**Reply by mbrasch** [reply]
The prompt should look something like this: `Environment: AAAA > username > path > git > `.

```toml
format = """${custom.henv} > $username$hostname > $directory > $git_branch $git_status > """
```

**Comment by mbrasch** [comment]
Has no one any ideas?

**Comment by mickimnet** [comment]
Hello @mbrasch,

Is the following, what you're looking for:
<img width="688" alt="Starship-Custom-HENV" src="https://github.com/starship/starship/assets/2911922/1a77acf2-6a1b-44dd-92f2-be006cac0149">

If yes, add the following to your configuration and edit it to your needs:
```toml
[custom.henv]
command = """HENV_VALUE="${HENV}"
case $HENV_VALUE in
AAAA)
  printf '\\033[31m%s\\033[0m' "AAAA"
  ;;
BBBB)
  printf '\\033[32m%s\\033[0m' "BBBB"
  ;;
CCCC)
  printf '\\033[33m%s\\033[0m' "CCCC"
  ;;
*)
  echo -n ""
  ;;
esac
"""
style = "fg:white bg:none"
when = true
shell = ["bash", "--noprofile", "--norc"]
format = "[$output]($style)"
```

You can use any string or symbol in the second parameter of `printf`. If you would like to use other colors, see the [list of ANSI color escape sequences](https://stackoverflow.com/questions/4842424/list-of-ansi-color-escape-sequences).

I hope that helps & take care
Mick

**Reply by mbrasch** [reply]
@mickimnet Thank you very much. That's it. :)

With format string like `'\\033[48;2;255;0;0m%s\\033[0m'` I get the background colour I need.

I had hoped that it would be a little more elegant via the starship style, but at least there is a workaround. And I was allowed to take a look into the (slightly disturbing) world of escape sequences. ;)

**Reply by mickimnet** [reply]
Glad I could help. And yes, the world of escape sequences is a bit disturbing because it is darn old 🤗
